src/lib.rs

```diff
@@ -178,6 +178,9 @@ pub fn derive_error(input: TokenStream) -> TokenStream {
                 Ok(msg) => display_matches.push_str(&msg),
             }
         }
+        display_matches.push_str(&format!(
+            "_ => unsafe {{ ::{std_crate}::hint::unreachable_unchecked()}}"
```

Comment:
I would strongly prefer not outputting `unsafe`. Especially when we have nothing preventing the unreachable code from actually being unreachable.

For instance, this is trivially unsound:

```rust
#[derive(Debug, onlyerror::Error)]
enum Error {
    /// Some
    Some,
    Other,
}

fn main() {
    println!("{}", Error::Other);
}
```

Reply:
Thanks for the feedback! I've missed that case.

## About parsing
I've fixed parsing of unit type in 2 steps:
1. When checking on variant type the token is not consumed if it is not a group
2. If variant type is unit we can just skip until ',' as otherwise the syntax will not e correct

## About unsafe
The `unreachable_unchecked` stuff is here as reference to std's derived impls for `Debug`, `Clone`, etc. I can replace it with `unreachable!()` if you prefer, this branch is obviously should be unreachable. It is here only because references are always considered inhabited.

Reply:
## About tests
I don't know how to test code that should not compile. So if you have examples I would like to see and write tests for this PR.

Reply:
Thanks! I was working on the underlying parsing issue in https://github.com/parasyte/myn/pull/8 -- I think what you have here is good for now, and we can update `myn` separately.

[`ui_test`](https://crates.io/crates/ui_test) is the usual way to test "this does not compile" cases along with assertions over patterns in the error messages.